cmd/kube-apiserver/app/server.go

```diff
@@ -232,6 +233,7 @@ func (s *APIServer) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&s.KubeletConfig.CertFile, "kubelet-client-certificate", s.KubeletConfig.CertFile, "Path to a client cert file for TLS.")
 	fs.StringVar(&s.KubeletConfig.KeyFile, "kubelet-client-key", s.KubeletConfig.KeyFile, "Path to a client key file for TLS.")
 	fs.StringVar(&s.KubeletConfig.CAFile, "kubelet-certificate-authority", s.KubeletConfig.CAFile, "Path to a cert. file for the certificate authority.")
+	fs.IntVar(&s.ApiServerNodePort, "apiserver-node-port", 0, "NodePort For load balanced api server endpoints.  Allows for highly available api server deployment behind a single URL.")
```

Comment:
As you suggested below, I just set 0 as an explicit flag to mark its unset.
We can use this as a signal to default behavior (api.ServiceTypeClusterIP)